### PR TITLE
Get discussions by created fix #768

### LINKED ISSUE #768: get_discussions_by_created return empty results on
**Steps to reproduce**
1. Create users:
```
create_account cyberfounder alice "{}" "300.000 GOLOS" true
create_account cyberfounder bob "{}" "300.000 GOLOS" true
create_account cyberfounder dave "{}" "300.000 GOLOS" true
```
2. Create posts from users:
```
post_comment alice test "" ptest "Test change1" "<h1>changed 1!</h1><hr>test edit" "" true
post_comment bob test "" ptest "Test change1" "<h1>changed 1!</h1><hr>test edit" "" true
post_comment dave test "" ptest "Test change1" "<h1>changed 1!</h1><hr>test edit" "" true
```
3. Send request to get list of discussions:
```
curl --data '{"method":"call","params":["tags", "get_discussions_by_created", [{}]],"jsonrpc":"2.0","id":0}' http://127.0.0.1:8090/rpc | python -mjson.tool  | grep -P '"author"|"permlink"'
....

"author": "dave",
"permlink": "test",
"author": "bob",
"permlink": "test",
"author": "alice",
"permlink": "test",
```
4. Send request to get list of discussions with offset:
```
curl --data '{"method":"call","params":["tags", "get_discussions_by_created", [{"start_author":"dave", "start_permlink":"test"}]],"jsonrpc":"2.0","id":0}' http://127.0.0.1:8090/rpc | python -mjson.tool  | grep -P '"author"|"permlink"'
....

```
**Actual result**
The last request returns empty results

**Expected result**
```
curl --data '{"method":"call","params":["tags", "get_discussions_by_created", [{"start_author":"dave", "start_permlink":"test"}]],"jsonrpc":"2.0","id":0}' http://127.0.0.1:8090/rpc | python -mjson.tool  | grep -P '"author"|"permlink"'
....

"author": "bob",
"permlink": "test",
"author": "alice",
"permlink": "test",
```

### PR DESCRIPTION
**Pull request by AKorpusenko**

### CODE REVIEW

File: plugins/tags/plugin.cpp [hunk 1]
```diff
@@ -490,11 +489,11 @@ namespace golos { namespace plugins { namespace tags {
...
                     const auto citr = cidx.find(query.start_comment.id);
-                    if (citr != cidx.end()) {
+                    if (citr == cidx.end()) {
                         return false;
                     }
-
-                    query.reset_start_comment();
+                    
+                    // query.reset_start_comment();
                     itr = idx.iterator_to(*citr);
```
Comment (afalaleev) [inline plugins/tags/plugin.cpp]: Restore line with `reset_start_comment()`
Add `++itr`

File: plugins/tags/plugin.cpp [hunk 2]
```diff
@@ -511,7 +510,6 @@ namespace golos { namespace plugins { namespace tags {
             }
             return true;
         });
-
```
Comment (afalaleev) [inline plugins/tags/plugin.cpp]: Restore this empty line back

File: plugins/tags/plugin.cpp [hunk 3]
```diff
@@ -413,7 +413,6 @@ namespace golos { namespace plugins { namespace tags {
             fill_discussion(d, query);
             d.hot = itr->hot;
             d.trending = itr->trending;
-
```
Comment (afalaleev) [inline plugins/tags/plugin.cpp]: Restore this line.

File: plugins/tags/plugin.cpp [hunk 4]
```diff
@@ -490,11 +489,11 @@ namespace golos { namespace plugins { namespace tags {
                 if (query.has_start_comment()) {
                     const auto& cidx = indices.get<tags::by_comment>();
                     const auto citr = cidx.find(query.start_comment.id);
-                    if (citr != cidx.end()) {
+                    if (citr == cidx.end()) {
                         return false;
                     }
-
-                    query.reset_start_comment();
+
```
Comment (afalaleev) [inline plugins/tags/plugin.cpp]: Remove this whitespaces.

File: plugins/tags/plugin.cpp [hunk 5]
```diff
@@ -522,7 +520,12 @@ namespace golos { namespace plugins { namespace tags {
         std::sort(it, et, DiscussionOrder());
 
         if (query.has_start_comment()) {
-            for (; et != it && it->id != query.start_comment.id; ++it);
```
Comment (afalaleev) [inline plugins/tags/plugin.cpp]: Restore this loop line.

Comment (AKorpusenko) [reply plugins/tags/plugin.cpp]: Done

File: plugins/tags/plugin.cpp [hunk 6]
```diff
@@ -492,9 +493,9 @@ namespace golos { namespace plugins { namespace tags {
                     if (citr == cidx.end()) {
                         return false;
                     }
-                    
-                    // query.reset_start_comment();
+                    query.reset_start_comment();
                     itr = idx.iterator_to(*citr);
+                    itr++;
```
Comment (afalaleev) [inline plugins/tags/plugin.cpp]: Change this to `++itr` - it works faster, then `itr++`

Comment (AKorpusenko) [reply plugins/tags/plugin.cpp]: Fixed that :hammer:. Thank you for the tip. Learned something new for myself

**Pull request merged**